src/chains.jl

```diff
@@ -671,9 +672,7 @@ Return a new `Chains` object with a `NamedTuple` type `n` placed in the `info` f
 new_chn = setinfo(chn, NamedTuple{(:a, :b)}((1, 2)))
 ```
 """
-function setinfo(c::Chains, n::NamedTuple)
-    return Chains(c.value, c.logevidence, c.name_map, n)
-end
+setinfo(c::Chains, n::NamedTuple) = Chains(c.value, c.logevidence, c.name_map, n, c.weights)
```

Comment:
Makes sense.

Reply:
Actually, I think this does raise the question of why `Chains` are immutable. I'm not sure there's many practical situations where we'll be creating enough `Chains` to mess with the garbage collecter, and there's nothing you can really stack-allocate anyways because everything is stuffed in an array. Is there something I'm missing?

Reply:
What would be the advantage of them being mutable? It seems it would make it easier to violate the constraints that we impose on the different fields. E.g., there are assumptions on the `range` of a `Chains` which could be broken easily. Since we don't copy but share the underlying data immutable `Chains` shouldn't cause (many) more allocations than a mutable alternative.